Python/ml/Polynomial_Regression.ipynb

```diff
@@ -0,0 +1,389 @@
+       "      <td>60000</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>Manager</td>\n",
+       "      <td>4</td>\n",
+       "      <td>80000</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>Country Manager</td>\n",
+       "      <td>5</td>\n",
+       "      <td>110000</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "            Position  Level  Salary\n",
+       "0   Business Analyst      1   45000\n",
+       "1  Junior Consultant      2   50000\n",
+       "2  Senior Consultant      3   60000\n",
+       "3            Manager      4   80000\n",
+       "4    Country Manager      5  110000"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df = pd.read_csv('Position_Salaries.csv')\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(array([[ 1],\n",
+       "        [ 2],\n",
+       "        [ 3],\n",
+       "        [ 4],\n",
+       "        [ 5],\n",
+       "        [ 6],\n",
+       "        [ 7],\n",
+       "        [ 8],\n",
+       "        [ 9],\n",
+       "        [10]]),\n",
+       " array([  45000,   50000,   60000,   80000,  110000,  150000,  200000,\n",
+       "         300000,  500000, 1000000]))"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "X = df.iloc[:,1:2].values\n",
+    "y = df.iloc[:,-1].values\n",
+    "X,y"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Splitting the dataset into train and test sets\n",
+    "#from sklearn.model_selection import train_test_split\n",
+    "#X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#Linear Regress to compare with Polynomial Regression\n",
+    "from sklearn.linear_model import LinearRegression\n",
+    "reg = LinearRegression()\n",
+    "reg.fit(X, y)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Polynomial Regression\n",
+    "from sklearn.preprocessing import PolynomialFeatures\n",
+    "poly_reg = PolynomialFeatures(degree=5)\n",
+    "X_poly = poly_reg.fit_transform(X ,y)"
```

Comment:
You have taken the degree of the polynomial regression as 5, as this was a small dataset it was easy to determine the degree.

I would like you to implement the same from the `car price prediction` dataset on Kaggle and also come up with a method to get the degree.

Hint - In the polynomial regression equation, which is the most significant term?